contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpAPIConfig.java

```diff
@@ -0,0 +1,135 @@
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import org.apache.drill.shaded.guava.com.google.common.base.MoreObjects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+
+@JsonTypeName("http-api-config")
+public class HttpAPIConfig {
+
+  private static final Logger logger = LoggerFactory.getLogger(HttpAPIConfig.class);
+
+  private final String url;
+
+  private final String method;
+
+  private final Map<String, String> headers;
+
+  private final String authType;
+
+  private final String username;
+
+  private final String password;
+
+  private final String postBody;
+
+  public HttpAPIConfig(@JsonProperty("url") String url,
+                       @JsonProperty("method") String method,
+                       @JsonProperty("headers") Map<String, String> headers,
+                       @JsonProperty("authType") String authType,
+                       @JsonProperty("username") String username,
+                       @JsonProperty("password") String password,
+                       @JsonProperty("postBody") String postBody) {
+    this.url = url;
+
+    // Get the request method.  Only accept GET and POST requests.  Anything else will default to GET.
+    if (method.toLowerCase().equals("get") || method.toLowerCase().equals("post")) {
+      this.method = method.toLowerCase();
+    } else {
+      this.method = "get";
+    }
+    this.headers = headers;
+
+    // Get the authentication method. Future functionality will include OAUTH2 authentication but for now
+    // Accept either basic or none.  The default is none.
+    this.authType = (authType == null || authType.isEmpty()) ? "none" : authType;
+    this.username = username;
+    this.password = password;
+    this.postBody = postBody;
+
+    // Validate the authentication type
+
+  }
+
+  @JsonProperty("url")
+  public String url() { return url; }
+
+  @JsonProperty("method")
+  public String method() { return method; }
+
+  @JsonProperty("headers")
+  public Map<String, String> headers() { return headers; }
+
+  @JsonProperty("authType")
+  public String authType() { return authType; }
+
+  @JsonProperty("username")
+  public String username() { return username; }
+
+  @JsonProperty("password")
+  public String password() { return password; }
+
+  @JsonProperty("postBody")
+  public String postBody() { return postBody; }
+
+  @Override
+  public int hashCode() {
+    return Arrays.hashCode(
+      new Object[]{url, method, headers, authType, username, password, postBody});
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+      .add("url", url)
+      .add("method", method)
+      .add("headers", headers)
+      .add("authType", authType)
+      .add("username", username)
+      .add("password", password)
+      .add("postBody", postBody)
+      .toString();
```

Comment:
This needs to be in Drill "DESCRIBE" format:

```
ClassName [prop=value, prop=value, ...]
```

Reply:
I replaced this and all other occurrences with the `PlanStringBuilder()` function